pyomo/core/base/component.py

```diff
@@ -1157,15 +1163,212 @@ def __ne__(self, other):
         except AttributeError:
             return self._cids.__ne__(other)
 
+    def _validate_slice_index(self, idx):
+        if idx.__class__ is not tuple:
+            idx = (idx,)
+        ellipsis_count = 0
+        slice_count = 0
+        fixed_count = 0
+        validated_index = []
+        for v in idx:
+            if type(v) == slice:
+                validated_index.append(v)
+                slice_count += 1
+            elif v == Ellipsis:
+                if ellipsis_count:
+                    raise NotImplementedError(
+                        "Got invalid index %s when creating CUID. "
+                        "Multiple ellipses are not supported." % (idx,)
+                        )
+                if fixed_count:
+                    raise NotImplementedError(
+                        "Got invalid index %s when creating CUID. "
+                        "Fixed indices are not supported in the same "
+                        "index as an ellipsis." % (idx,)
+                        )
```

Comment:
CUIDs do not support an ellipsis-like wildcard, `**`, with fixed indices as well.